client/state/ui/comments/reducer.js

```diff
@@ -67,4 +74,25 @@ export const queries = ( state = {}, action ) => {
 	}
 };
 
-export default combineReducers( { queries: keyedReducer( 'siteId', queries ) } );
+export const pendingActions = function( state = [], action ) {
+	switch ( action.type ) {
+		case COMMENTS_CHANGE_STATUS:
+		case COMMENTS_DELETE:
+			const key = getRequestKey( action );
+			if ( has( action, 'meta.dataLayer.trackRequest' ) && state.indexOf( key ) === -1 ) {
+				return [ ...state, key ];
+			}
+			return state;
+		case COMMENTS_LIST_REQUEST:
+		case COMMENTS_TREE_SITE_REQUEST:
```

Comment:
We don't query for tree that often (just once until we switch sites), so if folks are concerned about the list growing too long, I can also add an explicit clear action.